### PR TITLE
Horror Toy Factory

### PR DESCRIPTION
**Pull request by Braniff**
Appreciate any suggestions, thank you.

### CODE REVIEW

**Comment by Lemongrass3110** [comment]
Are you going to work on it and open it new later on?

**Comment by Jeybla** [comment]
Yes, he will ^^

**Comment by Atemo** [comment]
Ho well I'm working on it too.. almost finish too..

**Comment by Hismoon** [comment]
Bug's are fixed.
Appreciate any suggestions, thank you.

**Comment by lucasoli** [comment]
I think sc_onparam() can be replaced by getstatus() that is already coded and do the same thing. But correct me if I'm wrong.

**Comment by Hismoon** [comment]
@lucasoli You're right sir. I didn't found this command in my researchs and had coded with a friend the custom command. I can replace it if it's required.

**Comment by exneval** [comment]
I guess this is belong to refis?
you seem cut the original header

//= Auriga Script ==============================================================
// Ragnarok Online Toy Factory in Nightmare Script	by refis
//=============================================================

**Comment by Hismoon** [comment]
I had just copy the header earlier from a script in eathena forum because I found it pretty cool.

**Comment by Darkelfen** [comment]
I got curious here, because I dunno how the headings are done, well it is pretty obvious how it can be done, but for example how can I acknowledge the original creator of a file taken from another emulator and made compatible with rAthena?

Can you give us an example?

**Comment by secretdataz** [comment]
@Darkelfen 
```
//===== rAthena Script =======================================
//= Horror Toy Factory
//===== Description: =========================================
//= Blablablabla
//===== Changelogs: ==========================================
//= 1.0 First version [Refis]
//= 1.1 Translate and stuffs [Hismoon]
//= 1.2 Clean-up [JohnDoe]
//============================================================
```

**Comment by Lemongrass3110** [comment]
Additionally please either use a email adress that is attached to a github account or attach the one you used for these commits to your current account.

braniff

Actually rebasing and squashing these commits would be nice to have as well. ;-)

File: npc/re/instances/HorrorToyFactory.txt [hunk 1]
```diff
@@ -1607,13 +1607,15 @@ OnTimer5000:
 //=	Coordinator npc
 //============================================================
 1@xm_d,71,129,3	script	#fac1bs	10020,{/* 55928 */
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) == 1249) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) == 1249) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 2]
```diff
@@ -1607,13 +1607,15 @@ OnTimer5000:
...
+	if(getstatus(SC_MONSTER_TRANSFORM,1) == 1249) {
 		mes "[Coordinator]";
 		mes "What are you wearing?";
 		mes "I have a box here and they must go there!";
 		close;
 	}
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 3]
```diff
@@ -1653,11 +1655,13 @@ OnStart:
 //============================================================
 1@xm_d,65,127,6	script	#pck1	10033,{/* 55929 (hide)*/
 	progressbar "0x00FF00",1;
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) == 1249) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) == 1249) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 4]
```diff
@@ -1653,11 +1655,13 @@ OnStart:
...
+	if(getstatus(SC_MONSTER_TRANSFORM,1) == 1249) {
 		mes "^009900You are carrying the packaged present.^000000";
 		close;
 	}
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 5]
```diff
@@ -1680,7 +1684,8 @@ OnStart:
 		mes "Is it necessary to stop the workers, who are also active?^000000";
 		close;
 	}
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1249) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1249) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 6]
```diff
@@ -1706,7 +1711,8 @@ OnStart:
 //=	Warpportal to the second product line
 //============================================================
 1@xm_d,179,129,0	script	#fac2wp	45,2,2,{
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1249) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1249) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 7]
```diff
@@ -2876,7 +2882,8 @@ OnStart:
 1@xm_d,10,24,0	script	alert#61	139,10,10,{/* 56022 */
 OnAlertStart:
 OnTouch:
-	if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
+	//if(sc_onparam(SC_MONSTER_TRANSFORM,1) != 1246) {
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: npc/re/instances/HorrorToyFactory.txt [hunk 8]
```diff
@@ -3005,7 +3012,8 @@ OnTimer3000:
 }
 
 function	script	HTFWorker	{
-	if(sc_onparam(SC_MONSTER_TRANSFORM, 1) != 1246)
+	//if(sc_onparam(SC_MONSTER_TRANSFORM, 1) != 1246)
```
Comment (secretdataz) [inline npc/re/instances/HorrorToyFactory.txt]: Remove commented line

File: src/custom/script.inc [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+ *		<code>;
+ *	}
+ **/
+
+//BUILDIN_FUNC(example)
+//{
+//	ShowInfo("buildin_example: It works!\n");
+//	script_pushint(st,1);
+//	return 0;
+//}
```
Comment (secretdataz) [inline src/custom/script.inc]: Missing EOF

File: src/custom/script_def.inc [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+// For more information, see LICENCE in the main folder
+
+/**
+ * Custom Script Commands
+ * Place the definition of custom script commands in this file.
+ * Format:
+ *	BUILDIN_DEF(command_name,"parameters"),
+ **/
+
+//BUILDIN_DEF(example,""),
```
Comment (secretdataz) [inline src/custom/script_def.inc]: Missing EOF

**Comment by Hismoon** [comment]
Squashed and hopefully better.
Any other suggestions are welcome.

**Comment by lucasoli** [comment]
@Hismoon Looks great! Maybe you should just adjust the header according to the other instance scripts.

```
//===== rAthena Script =======================================
//= Horror Toy Factory
//===== By: ==================================================
//= Braniff
//===== Current Version: =====================================
//= 1.1
//===== Compatible With: =====================================
//= rAthena Project
//===== Description: =========================================
//= Solve the mystery about the toy factory and what happened
//= to Celine Kimi and the grandpa.
//===== Additional Comments: =================================
//= 1.0 First version. [Unknown]
//= 1.1 Translate and add functionality to script. [Braniff]
//============================================================
```

Thank you for sharing this!

**Comment by Lemongrass3110** [comment]
@lucasoli the example @secretdataz stated was perfectly fine and is the header we agreed to use/change whenever we adjust a script in the future.
Current version and by are both duplicates from the last entry in the changelog and it is stupid to maintain it in two places all the time.
We also agreed that we are going to drop the compatible with section of the script, because the headline already says it is an rAthena script and that is the only thing we guarantee compatability in our github repository.

**Comment by lucasoli** [comment]
@Lemongrass3110 That header was taken from OctopusCave script, I thought that it was equals to the one @secretdataz suggested, I'm sorry for that. And I agree with the other changes that you've comment.

**Comment by Braniff** [comment]
Ready for merge if no other things are necessary.

**Comment by Atemo** [comment]
Thing is @esu1214 shared the official structure for this script http://herc.ws/board/topic/13378-instancehorror-toy-factory-aegis-server/
and you based your PR on auriga which we don't have clue about the source.
There are notables differences like BaseLevel min to enter (110 in your, 140 on the other - it seems to be 140 on official), the viewpoints added in auriga and in the aegis file we have the hp isn't scaled with the count of party member (which is wrong? need more info).

My personnal opinion (and not because i made it on my local thx) : i prefer a work based on the aegis files. But i want the input of others because you made a huge work and i'm not the one to decide.

**Comment by Braniff** [comment]
The BaseLevel was from the script I had from Auriga. The other things i.e. scale HP are from iRO.  
Link to my source: http://irowiki.org/wiki/Horror_Toy_Factory

**Comment by Jeybla** [comment]
@Atemo can you open a PR with your current aegis script? If something is missing, please provide a todo list, too.
It'd be better to compare both, thanks!

**Comment by Atemo** [comment]
@Jeybla https://github.com/rathena/rathena/compare/horror_toy_factory

**Comment by Braniff** [comment]
After the overlook of the PR from @Atemo I have some points figured out.

Positive:
+ implentation from the skill "MAXPAIN"
+ more and better information from iRO server i.e. npc messages or announces

Negative:
- hp scaling with each party member

The HP scaling can be added in further updates, but it should be in the script.

**Comment by Atemo** [comment]
From my test on iRO, the HP of the boss doesn't increase with the count of party member in the instance (like in the aegis script). irowiki source..?
On this screenshot, 3 members in the instance :
screenchaos129

**Comment by Braniff** [comment]
http://irowiki.org/wiki/Horror_Toy_Factory
> She has a large spongy Max HP (20m ~ 66m, depends on party size, 20m if solo, 66m if party of 12), high def, and high mdef, so it's best to have high ASPD AoE skills to take her down.

**Comment by lucasoli** [comment]
@Atemo try to repeat the instance with 4 members on your party. Celine has a fixed HP (20.000.000) if the party have up to 4 members, but she scale 6.000.000 HP by each member that joins the party after the 4th member.

Source:
http://sites.levelupgames.uol.com.br/forum/ragnarok/showthread.php?164933-F%E1brica-do-Terror-Bioquimicos-MVP-Celine-Kimi

I'm sorry for my bad english.

**Comment by Atemo** [comment]
We appreciate your huge work however we prefer a script converted from aegis leak.

**Pull request closed without merging**